pkg/build/controller/factory/factory.go

```diff
@@ -261,6 +260,45 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 	}
 }
 
+type BuildConfigControllerFactory struct {
+	Client                  osclient.Interface
+	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	// Stop may be set to allow controllers created by this factory to be terminated.
+	Stop <-chan struct{}
+}
+
+// Create creates a new ConfigChangeController which is used to trigger builds on creation
+func (factory *BuildConfigControllerFactory) Create() controller.RunnableController {
+	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	cache.NewReflector(&buildConfigLW{client: factory.Client}, &buildapi.BuildConfig{}, queue, 2*time.Minute).RunUntil(factory.Stop)
+
+	bcController := &buildcontroller.BuildConfigController{
+		BuildConfigInstantiator: factory.BuildConfigInstantiator,
+	}
+
+	return &controller.RetryController{
+		Queue: queue,
+		RetryManager: controller.NewQueueRetryManager(
+			queue,
+			cache.MetaNamespaceKeyFunc,
+			func(obj interface{}, err error, retries controller.Retry) bool {
+				bc := obj.(*buildapi.BuildConfig)
+				if maxRetries > retries.Count {
+					glog.V(3).Infof("Giving up retrying BuildConfig update event %s/%s: %v", bc.Namespace, bc.Name, err)
+					kutil.HandleError(err)
+					return false
+				}
+				glog.V(4).Infof("Retrying BuildConfig update event %s/%s: %v", bc.Namespace, bc.Name, err)
+				return true
```

Comment:
If something else already kicked off the first build, a new buildconfig entry comes up in the queue with the updated buildconfig resulting in a no more retries. I couldn't think of something that would cause unnecessary retries, but we can discuss.